src/libxtp/bulkesp.cc

```diff
@@ -0,0 +1,768 @@
+/* 
+ *            Copyright 2009-2018 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+
+#include <votca/xtp/bulkesp.h>
+#include <votca/xtp/aomatrix.h>
+#include <votca/tools/linalg.h>
+//#include <boost/progress.hpp>
+#include <votca/xtp/numerical_integrations_periodic.h>
+#include <math.h> 
+#include <votca/tools/constants.h>
+
+#include "votca/xtp/orbitals.h"
+
+#include <fstream>
+
+using namespace votca::tools;
+
+
+namespace votca {
+    namespace xtp {
+        namespace ub = boost::numeric::ublas;
+        
+
+        std::vector<Bulkesp::Molecule> Bulkesp::BreakIntoMolecules(std::vector< ctp::QMAtom* > _atoms, double scale) {
+
+            std::vector<Bulkesp::Molecule> mols;
+            std::list<Bulkesp::Bond> bonds;
+
+            Elements _elements;
+
+            CTP_LOG(ctp::logDEBUG, *_log) << " BreakIntoMolecules(): locating bonds.\n" << flush;
+            
+            //find all the bonds;
+            for (vector<ctp::QMAtom*>::iterator i = _atoms.begin(); i != _atoms.end(); ++i) {
+                for (vector<ctp::QMAtom*>::iterator j = i + 1; j != _atoms.end(); ++j) {
+                    tools::vec dif;
+                    if(periodic){
```

Comment:
why this if clause?

Reply:
Should have been removed when wrapping was commented out